File: 단계별로 풀어보기/1차원 배열/4344/평균은 넘겠지.cpp
#include <iostream>
using namespace std;

double* makeGradeList(double* gradeList,int stuCnt) {

	for (int i = 0; i < stuCnt; i++) {
		cin >> gradeList[i];
	}

	return gradeList;
}

double calculateAvg(double* gradeList, int stuCnt) {
	double total = 0.0;

	for (int i = 0; i < stuCnt; i++) {
		total+=gradeList[i];
	}

	return total / (double)stuCnt;
}

double calculateRate(double* gradeList, int stuCnt, double avg) {
	int cnt = 0;

	for (int i = 0; i < stuCnt; i++) {
		if (avg < gradeList[i])
			cnt++;
	}

	return ((double)cnt) / ((double)stuCnt)*(double)100; //형변환
}

double findOverAvgRate() {
	int stuCnt;
	cin >> stuCnt;
	double* gradeList = new double[stuCnt];

	cout << fixed; //소수점 고정
	cout.precision(3);

	makeGradeList(gradeList, stuCnt);
	double avg = calculateAvg(gradeList, stuCnt);
	double rate = calculateRate(gradeList, stuCnt, avg);

	delete gradeList;

	return rate;
}

int main() {
	ios_base::sync_with_stdio(false);
	cin.tie(NULL);

	int caseNum;
	cin >> caseNum;

	double* caseList = new double[caseNum];

	for (int i = 0; i < caseNum; i++) {
		caseList[i] = findOverAvgRate();
	}

	for (int i = 0; i < caseNum; i++) {
		cout << caseList[i] << '%'<<'\n';
	}

	delete caseList;

	return 0;
}

File: 단계별로 풀어보기/1차원 배열/4344/평균은 넘겠지.py
import sys

def makeGradeList():
    return list(map(float,sys.stdin.readline().split()))

def calculateAvg(gradeList):
    total = 0.0
    for grade in gradeList:
        total+=grade
        
    total -= gradeList[0]

    return float(total)/(len(gradeList)-1)

def findOverAvgRate():
    gradeList=makeGradeList()
    avg=calculateAvg(gradeList)
    cnt=0

    
    for i in range(1,len(gradeList)):
        if avg < gradeList[i]:
            cnt+=1

    return float(cnt)/(len(gradeList)-1)*100    

def init():
    C=int(sys.stdin.readline())
    caseResult=[]
    for i in range(C):
        caseResult.append("{:.3f}" .format(findOverAvgRate())+'%')
    for result in caseResult:
        print(result)

init()
